crates/pathfinder/src/sierra.rs

```diff
@@ -11,40 +10,111 @@ pub const COMPILER_VERSION: &str = env!("SIERRA_CASM_COMPILER_VERSION");
 /// The class representation expected by the compiler doesn't match the representation used
 /// by the feeder gateway for Sierra classes, so we have to convert the JSON to something
 /// that can be parsed into the expected input format for the compiler.
-pub fn compile_to_casm(sierra_definition: &[u8]) -> anyhow::Result<Vec<u8>> {
-    let feeder_gateway_class_definition =
-        serde_json::from_slice::<FeederGatewayContractClass<'_>>(sierra_definition)
-            .context("Parsing Sierra class")?;
-    let sierra_class: ContractClass = feeder_gateway_class_definition
-        .try_into()
-        .context("Converting to Sierra class")?;
-
-    validate_compatible_sierra_version(
-        &sierra_class,
-        ListSelector::ListName(
-            cairo_lang_starknet::allowed_libfuncs::DEFAULT_EXPERIMENTAL_LIBFUNCS_LIST.to_string(),
-        ),
-    )
-    .context("Validating Sierra class")?;
-
-    let casm_class =
-        CasmContractClass::from_contract_class(sierra_class, true).context("Compiling to CASM")?;
-    let casm_definition = serde_json::to_vec(&casm_class)?;
-
-    Ok(casm_definition)
+pub fn compile_to_casm(
+    sierra_definition: &[u8],
+    version: &StarknetVersion,
+) -> anyhow::Result<Vec<u8>> {
+    let definition = serde_json::from_slice::<FeederGatewayContractClass<'_>>(sierra_definition)
+        .context("Parsing Sierra class")?;
+
+    const V_0_11_0: semver::Version = semver::Version::new(0, 11, 0);
+
+    match version
+        .parse_as_semver()
+        .context("Deciding on compiler version")?
```

Comment:
This will block sync progress if the parsing fails. I'm not sure we can really do any different though.